blocks/carousel/carousel.js

```diff
@@ -0,0 +1,159 @@
+  + '</svg>';
+const HTML_ARROW_RIGHT = '<svg fill="rgba(238, 238, 238, 0.9)" xmlns="http://www.w3.org/2000/svg" width="24px" height="24px" viewBox="0 0 600 600">\n'
+  + '<path d="M345.441,248.292L151.154,442.573c-12.359,12.365-32.397,12.365-44.75,0c-12.354-12.354-12.354-32.391,0-44.744L278.318,225.92L106.409,54.017c-12.354-12.359-12.354-32.394,0-44.748c12.354-12.359,32.391-12.359,44.75,0l194.287,194.284c6.177,6.18,9.262,14.271,9.262,22.366C354.708,234.018,351.617,242.115,345.441,248.292z"/>\n'
+  + '</svg>\n';
+
+const activateBullet = (bulletId) => {
+  const btnNav = document.querySelector('.bottom-nav');
+  [...btnNav.children].forEach((el) => {
+    if (el.id === bulletId) {
+      el.classList.replace('inactive', 'active');
+    } else {
+      el.classList.replace('active', 'inactive');
+    }
+  });
+};
+
+const activateSlide = (targetPicture) => {
+  // Switch slide pictures
+  const slider = document.querySelector('.slider');
+  [...slider.children].forEach((el) => {
+    if (el.id === targetPicture) {
+      el.classList.replace('hide', 'show');
+    } else {
+      el.classList.replace('show', 'hide');
+    }
+  });
+};
+
+const bottomNavigation = (event) => {
+  const target = event.target.id;
+  activateSlide(target);
+  activateBullet(target);
+};
+
+const arrowNavigation = (event) => {
+  const navButton = event.currentTarget.id;
+  let nextSlide = 0;
+  if (navButton === 'arrow-left') {
+    nextSlide = (activeSlide - 1) > 0 ? activeSlide - 1 : slideCount;
+  } else if (navButton === 'arrow-right') {
+    nextSlide = (activeSlide + 1) <= slideCount ? activeSlide + 1 : 1;
+  }
+  activateSlide(`slide-${nextSlide}`);
+  activateBullet(`slide-${nextSlide}`);
+  activeSlide = nextSlide;
+};
+
+const bottomNavOnClickEvents = () => {
+  const btnNav = document.querySelector('.bottom-nav');
+  [...btnNav.children].forEach((el) => {
+    el.addEventListener('click', bottomNavigation);
+  });
+};
+
+const arrowNavOnClickEvents = () => {
+  const arrowNav = document.querySelector('.arrow-nav');
+  [...arrowNav.children].forEach((el) => {
+    el.addEventListener('click', arrowNavigation);
+  });
+};
+
+// Control Slide show
+const toNextSlide = () => {
+  const nextSlide = (activeSlide + 1) <= slideCount ? activeSlide + 1 : 1;
+
+  activateSlide(`slide-${nextSlide}`);
+  activateBullet(`slide-${nextSlide}`);
+  activeSlide = nextSlide;
+};
+
+// Start slide show
+const startSlideShow = () => {
+  if (!slideShow) {
+    slideShow = window.setInterval(toNextSlide, 5000);
+  }
+};
+
+const stopSlideShow = () => {
+  slideShow = window.clearInterval(slideShow);
+};
+
+const initSlider = () => {
+  const sliderWrapper = document.querySelector('.slider-wrapper');
+  sliderWrapper.addEventListener('mouseover', stopSlideShow);
+  sliderWrapper.addEventListener('mouseleave', startSlideShow);
+  window.addEventListener('load', startSlideShow);
+};
+
+export default function decorate(block) {
```

Comment:
i would encapsulate thematic lines of codes into functions, e.g.
- appendPictures
- setPictureIDs
- createArrowNavigation
- ...